financial-connections/src/main/java/com/stripe/android/financialconnections/ui/components/MiddleEllipsisText.kt

```diff
@@ -0,0 +1,210 @@
+package com.stripe.android.financialconnections.ui.components
+
+import androidx.compose.material.LocalTextStyle
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.SubcomposeLayout
+import androidx.compose.ui.text.TextLayoutResult
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.text.font.FontStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.text.style.TextDecoration
+import androidx.compose.ui.unit.Constraints
+import androidx.compose.ui.unit.TextUnit
+import java.text.BreakIterator
+
+/**
+ * A [Text] composable that shows the middle part of the text if the text is too long to fit in the
+ * given constraints.
+ *
+ * Note: Middle ellipsis text is not yet supported in Compose. This is a temporary solution until
+ * the official support is added. see this issue for more details: https://issuetracker.google.com/issues/185418980
+ *
+ **/
+@Suppress("LongMethod")
+@Composable
+internal fun MiddleEllipsisText(
```

Comment:
this seems very useful for other components as well, should we put it into stripe-ui-core?